aikau/src/main/resources/alfresco/upload/UploadTarget.js

```diff
@@ -0,0 +1,179 @@
+         var responseTopic = this.generateUuid();
+         this._uploadSubHandle = this.alfSubscribe(responseTopic, lang.hitch(this, this.onFileUploadComplete), true);
+
+         // To avoid the issue with processing payloads containing files with native
+         // code in them, it is necessary to temporarily store the files in the data model...
+         var filesRef = this.generateUuid();
+         this.alfSetData(filesRef, files);
+
+         this.alfPublish("ALF_CREATE_FORM_DIALOG_REQUEST", {
+            dialogId: "ALF_UPLOAD_TARGET_DIALOG",
+            dialogTitle: "upload.target.dialog.title",
+            dialogConfirmationButtonTitle: "upload.target.dialog.confirmation",
+            dialogCancellationButtonTitle: "upload.target.dialog.cancellation",
+            formSubmissionTopic: "ALF_UPLOAD_REQUEST",
+            formSubmissionPayloadMixin: {
+               alfResponseTopic: responseTopic,
+               filesRefs: filesRef,
+               targetData: uploadConfig
+            },
+            fixedWidth: true,
+            widgets: lang.clone(this.widgetsForLocationSelection)
+         }, true);
+      },
+
+      /**
+       * Sets up the widgets DOM element with file drag-and-drop upload capabilities.
+       * 
+       * @instance
+       */
+      postCreate: function alfresco_upload_UploadTarget__postCreate() {
+         this.inherited(arguments);
+         this.addUploadDragAndDrop(this.domNode);
+         this.addNodeDropTarget(this.domNode);
+         
+         // NOTE: A _currentNode needs to be set in order to avoid exceptions
+         this._currentNode = {
+            nodeRef: null
+         };
+
+         this.processWidgets(lang.clone(this.widgetsForButton), this.buttonNode);
+      },
+
+      /**
+       * The widget model for providing the user with an interface to be displayed in a form dialog
+       * for selecting the location to upload files to when they are dragged and dropped onto the 
+       * widget.
+       * 
+       * @instance
+       * @type {object[]}
+       */
+      widgetsForLocationSelection: [
+         {
+            name: "alfresco/forms/controls/ContainerPicker",
+            config: {
+               id: "FOLDER_PICKER",
+               label: "upload.target.location.picker.label",
+               description: "upload.target.location.picker.description",
+               name: "targetData.destination",
+               requirementConfig: {
+                  initialValue: true
+               }
+            }
+         }
+      ],
+
+      /**
+       * A widget model for requesting both the files to upload as well as the location to upload
+       * them to. By default the widget model is just a basic [AlfButton]{@link module:alfresco/buttons/AlfButton}.
+       * 
+       * @instance
+       * @type {object[]}
+       */
+      widgetsForButton: [
+         {
+            name: "alfresco/buttons/AlfButton",
+            config: {
```

Comment:
You can't really do this here, because you could conceivably have two instances on the same page and this would cause a problem. The button can be referenced easily in test automation (as we are doing using a combination of the UploadTarget ID and the button). Also, you you override this widget model if someone really wanted for their own tests.